source/ptreferential/ptreferential_utils.cpp

```diff
@@ -142,14 +142,15 @@ Indexes get_impacts_by_tags(const std::vector<std::string>& tag_name, const Data
     return result;
 }
 
-static bool keep_vj(const type::VehicleJourney* vj, const bt::time_period& period) {
+static bool keep_vj(const type::VehicleJourney* vj, const bt::time_period& period, const type::RTLevel rt_level) {
     if (vj->stop_time_list.empty()) {
         return false;  // no stop time, so it cannot be valid
     }
 
     const auto& first_departure_dt = vj->earliest_time();
     for (boost::gregorian::day_iterator it(period.begin().date()); it <= period.last().date(); ++it) {
-        if (!vj->base_validity_pattern()->check(*it)) {
+        if (!(vj->base_validity_pattern()->check(*it)
```

Comment:
You can acces the VP with the flatmap: `flat_enum_map<RTLevel, ValidityPattern*> validity_patterns = {{{nullptr, nullptr, nullptr}}};`